Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```
@@ -39,54 +39,61 @@ class Xcodeproj {
 
 // Convenience methods and properties
 extension Xcodeproj {
-    /// Builds an `Xcodeproj` instance by parsing the an `.xcodeproj` or `pbxproj` file at the provided path
+    /// Builds an `Xcodeproj` instance by parsing the `.xcodeproj` or `pbxproj` file at the provided path
     convenience init(path: String) throws {
         try self.init(url: URL(fileURLWithPath: path))
     }
-
-    /// The directory where the `.xcodeproj` resides. 
+    
+    /// The directory where the `.xcodeproj` resides.
     var projectDirectory: URL { projectURL.deletingLastPathComponent().deletingLastPathComponent() }
     /// The list of `PBXNativeTarget` targets in the project. Convenience getter for `PBXProjFile.nativeTargets`
     var nativeTargets: [PBXNativeTarget] { pbxproj.nativeTargets }
     /// The list of `PBXBuildFile` files a given `PBXNativeTarget` will build. Convenience getter for `PBXProjFile.buildFiles(for:)`
     func buildFiles(for target: PBXNativeTarget) -> [PBXBuildFile] { pbxproj.buildFiles(for: target) }
-
+    
     /// Finds the full path / URL of a `PBXBuildFile` based on the groups it belongs to and their `sourceTree` attribute
-    func resolveURL(to buildFile: PBXBuildFile) -> URL {
-        do {
-            let fileRef = try self.pbxproj.object(id: buildFile.fileRef) as PBXFileReference
-            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
-        } catch {
-            // Cover `XCVersionGroup` (like `*.xcdatamodel`) and `PBXVariantGroup` (like `*.strings`)
-            let fileRef = try! self.pbxproj.object(id: buildFile.fileRef) as PBXGroup
-            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
+    func resolveURL(to buildFile: PBXBuildFile) throws -> URL? {
+        if let fileRefID = buildFile.fileRef, let fileRefObject = try? self.pbxproj.object(id: fileRefID) as PBXFileReference {
+            return try resolveURL(objectUUID: fileRefID, object: fileRefObject)
+        } else {
```

`PBXBuildFile.fileRef` is now an optional (`ObjectUUID?`) to cover for the case of `PBXBuildFile` objects in the `.pbxproj` pointing to things other than a `fileRef`.

One of such cases are `PBXBuildFile` objects pointing to a `productRef` (e.g. a Swift Package dependency) instead of a `fileRef`.

We want to silently ignore the case of those type of `PBXBuildFiles` (as opposed to crashing during decoding of the plist, or trying to implement `struct PBXBuildFile` to parse them while we don't even have a need to handle those), and for those cases just make `resolveURL` return `nil`, hence the `if let fileRefID = buildFileRef` here now.